### PR TITLE
Add choice providers

### PR DESCRIPTION
**Pull request by dannyroberts**
closing https://github.com/dimagi/commcare-hq/pull/9355 in favor of this

### CODE REVIEW

File: corehq/apps/es/es_query.py [hunk 1]
```diff
@@ -302,6 +302,31 @@ def remove_default_filter(self, default):
...
+            return self.fields(fields).run().hits
+        else:
+            return self.run().hits
+
+    def values_list(self, *fields, **kwargs):
+        """modeled after django's QuerySet.values_list"""
+        flat = kwargs.pop('flat', False)
+        if kwargs:
+            raise TypeError('Unexpected keyword arguments to values_list: %s'
+                            % (list(kwargs),))
```
Comment (esoergel) [inline corehq/apps/es/es_query.py]: That's annoying - think it's the only way to do it though.

File: corehq/apps/es/es_query.py [hunk 2]
```diff
@@ -302,6 +302,31 @@ def remove_default_filter(self, default):
...
+            raise TypeError("'flat' is not valid when values_list is called with more than one field.")
+        if not fields:
+            raise TypeError('must be called with at least one field')
+        if flat:
+            field, = fields
+            return [hit[field]
+                    for hit in self.fields(fields).run().hits]
+        else:
+            return [tuple(hit[field] for field in fields)
+                    for hit in self.fields(fields).run().hits]
```
Comment (esoergel) [inline corehq/apps/es/es_query.py]: Looks good to me!

**Comment by czue** [comment]
looks good! didn't have any blockers. I agree that @esoergel should review the ES stuff. :+1: to merge from my side though

File: corehq/apps/es/es_query.py (hunk 2 above)
Comment (dannyroberts) [inline corehq/apps/es/es_query.py]: Thanks for the review!

**Comment by dannyroberts** [comment]
:white_check_mark:

**Comment by dannyroberts** [comment]
self-merging based on cory's comment

**Pull request merged**